sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2184,7 +2184,7 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
    * Create a Spark DataType.
    */
   private def visitSparkDataType(ctx: DataTypeContext): DataType = {
-    HiveStringType.replaceCharType(typedVisit(ctx))
+    HiveVoidType.replaceVoidType(HiveStringType.replaceCharType(typedVisit(ctx)))
```

Comment:
I don't get it why we need `HiveVoidType`. What happens if we just parse `void` to `NullType`?

Reply:
Because that could indicate VOID is a Hive type, the handle processing is more unified. Or, we can just use the PR #28833

Reply:
For example,  below function will point the failure is due to the legacy hive void type. If we mix VOID and NULL, I am not sure it would be better than separation.
```scala
  def failVoidType(dt: DataType): Unit = {
    if (HiveVoidType.containsVoidType(dt)) {
      throw new AnalysisException(
        "Cannot create tables with Hive VOID type.")
    }
  }
```

Reply:
VOID and NULL are indeed the same type. We can just check null type and fail with error message: `Cannot create tables with VOID type`

Reply:
The point is consistency: The VOID type in SQL statement should be the same as `NullType` specified by Scala API in `spark.catalog.createTable`.

Reply:
@cloud-fan , ok. I will follow your suggestion to fix it in #28833 , since this PR is a refator with new type `HiveVoidType`. Now we don't need it.